### ISSUE TITLE
Question: Do profiles have access to be able to create dialogs?

### ISSUE THREAD

**Issue by robert914** [3noix/UJPS#17]
I have a curious question. I haven't used Qt in C++ much, but I know basically how various widgets, dialogs and windows can be created. My question is more, do we have access to the root Qt application window such that our Profiles can pop-up a dialog, window, etc. to query the user of information?

The reason I ask is, the profile I'm working on for Star Citizen will support either GamePad or HOTAS mapping from the same profile. Currently, I have the profile code change what it maps based on what devices are plugged in. This works fine for my use, as I make it default to one mode if by chance I have all devices plugged in.

However, I got to thinking, if for some reason I had all my devices plugged in at the time, but I wanted to select which mode to run in at startup, it would be cool if I could pop-up a dialog asking which mode to use when either would be valid.

Is this possible? Seems like I'd need access to the root Qt object of the application, but I don't think the profiles have access to that, or do they?

**Comment by 3noix** [comment]
Yes, it is totally possible. The only thing is that you will need to use a modal window so that it blocks the timer (that calls the profile "run" function at each time step). To do so you will have to:
- add "QT += gui widgets" in the .pro file and your added source files to HEADERS and SOURCES
- code your window. I suggest you to sub-class QDialog. I will send you an example by email
- in "runFirstStep":
```
void Profile::runFirstStep()
{
	//...
	
	if (thereIsAChoiceToDo)
	{
		MyDialog dialog;
		dialog.exec();
		bool bGamePad = dialog.useGamepad();
	}
	
	//...
}
```
